### ISSUE TITLE
Delete with block

### ISSUE THREAD

**Issue by paoloandrea** [willowtreeapps/WTAData#39]
There exists the possibility to have a block for deleting a single entity and other for deleting all entities present in the database?
If so, how?
Thank you

**Comment by eriklamanna** [comment]
Yes, look in NSManagedObject+WTAData.h for ```+ (void)deleteAllInContext:(NSManagedObjectContext *)context;```.  There is also a delete all function that takes a predicate as well.  Wrap that in a save block for the context you want to use and it that should do what you want.

**Comment by paoloandrea** [comment]
But.. another question:
Is possible have a function that will return if the single or all entities are deleted?
eg:

- (void)deleteBlock:(void (^)(NSManagedObjectContext *context))work
       completion:(nullable void (^)(BOOL savedChanges, NSError * __nullable error))completion;
and

- (void)deleteAllBlock:(void (^)(NSManagedObjectContext *context))work
       completion:(nullable void (^)(BOOL savedChanges, NSError * __nullable error))completion;

**Comment by eriklamanna** [comment]
So the intended design is that the category functions on NSManagedObject are what are used to manipulate any of the managed objects.  They are intentionally kept light-weight so to be flexible in many different cases.  So in your case where you want to delete all of the objects of a certain type, you could do the following:

```
[wtaDataInstance saveInBackground:^(NSManagedObjectContext *context) {
    [Entity deleteAllInContext:context];
 } completion:^(BOOL savedChanges, NSError *error) {
     NSLog(@"All entities were deleted");
 }];
```
